src/edu/psu/sweng500/team8/coreDataStructures/Cell.java

```diff
@@ -32,10 +32,10 @@ public Cell(int rowIndex, int columnIndex) {
 	}
 	
 	public Cell(Cell cellToCopy) {
-		m_coordinates = cellToCopy.m_coordinates;
+		m_coordinates = new CellCoordinates(cellToCopy.m_coordinates.getRowIndex(), cellToCopy.m_coordinates.getColumnIndex());

```

Comment:
during testing, undo/redo was malfunctioning if multiple undos were done. while fixing a series of things, this was changed to make sure the copy is done correctly.